owncloudApp/src/main/java/com/owncloud/android/usecases/transfers/uploads/RetryUploadFromSystemUseCase.kt

```diff
@@ -16,36 +18,36 @@
  * You should have received a copy of the GNU General Public License
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
+
 package com.owncloud.android.usecases.transfers.uploads
 
-import android.content.Context
-import androidx.work.WorkManager
-import com.owncloud.android.datamodel.UploadsStorageManager
 import com.owncloud.android.domain.BaseUseCase
 import com.owncloud.android.domain.files.model.OCFile.Companion.PATH_SEPARATOR
+import com.owncloud.android.domain.transfers.TransferRepository
+import com.owncloud.android.domain.transfers.model.TransferStatus
 
 class RetryUploadFromSystemUseCase(
-    private val context: Context,
+    private val uploadFilesFromSystemUseCase: UploadFilesFromSystemUseCase,
+    private val transferRepository: TransferRepository,
 ) : BaseUseCase<Unit, RetryUploadFromSystemUseCase.Params>() {
 
     override fun run(params: Params) {
 
-        val uploadsStorageManager = UploadsStorageManager(context.contentResolver)
-        val failedUploads = uploadsStorageManager.failedUploads
-        val filteredUploads = failedUploads.filter { it.uploadId == params.uploadIdInStorageManager }
+        val failedUploads = transferRepository.getFailedTransfers()
+        val filteredUploads = failedUploads.filter { it.id == params.uploadIdInStorageManager }
         val uploadToRetry = filteredUploads.firstOrNull()
 
         uploadToRetry ?: return
 
-        val workManager = WorkManager.getInstance(context)
-        UploadFilesFromSystemUseCase(workManager).execute(
+        uploadFilesFromSystemUseCase.execute(
             UploadFilesFromSystemUseCase.Params(
                 accountName = uploadToRetry.accountName,
                 listOfLocalPaths = listOf(uploadToRetry.localPath),
                 uploadFolderPath = uploadToRetry.remotePath.trimEnd(PATH_SEPARATOR),
             )
         )
-        uploadsStorageManager.updateUpload(uploadToRetry.apply { uploadStatus = UploadsStorageManager.UploadStatus.UPLOAD_IN_PROGRESS })
+
+        transferRepository.updateTransfer(uploadToRetry.apply { status = TransferStatus.TRANSFER_IN_PROGRESS })
```

Comment:
You seem to be right. I kept it because it was before there and conceptually, it's more or less the same, but now thinking about that it's true that there's no evidence that it is necessary 👍